overlord/ifacestate/helpers.go

```diff
@@ -520,3 +522,54 @@ func snapsWithSecurityProfiles(st *state.State) ([]*snap.Info, error) {
 
 	return infos, nil
 }
+
+// SnapInterfaceState describes state of interfaces of a single snap.
+type SnapInterfaceState struct {
+	// Revision is the revision of the snap that was loaded into the
+	// interface repository. When snapd restarts it may have added a snap to
+	// the interface repository even when that snap was not yet active. To
+	// ensure that we can reload the interface repository correctly, for
+	// inactive snaps, this is the revision to load on startup.
+	Revision snap.Revision `json:"revision"`
+}
+
+// ifaceRepoKey is the key used to store state of the interface repository in the snapd state.
+const ifaceRepoKey = "interfaces"
```

Comment:
this is a strange name for a map that is keyed by snap names,  "snap-interfaces"  or "snap-profiles"  given that is also the bookkeeping in state around per-snap profiles

Reply:
Let's try `snap-interfaces`.